pydantic_ai_slim/pydantic_ai/models/openai.py

```diff
@@ -299,12 +307,29 @@ def _process_response(self, response: chat.ChatCompletion) -> ModelResponse:
         timestamp = datetime.fromtimestamp(response.created, tz=timezone.utc)
         choice = response.choices[0]
         items: list[ModelResponsePart] = []
+        vendor_metadata: dict[str, Any] = {}
+
+        # Add logprobs to vendor_metadata if available
+        if choice.logprobs is not None and choice.logprobs.content:
+            # Convert logprobs to a serializable format
+            vendor_metadata['logprobs'] = [
```

Comment:
Can you make sure `vendor_metadata` is passed as `None` if there are no logprobs, so that we don't unnecessarily clutter the model response?